index.js

```diff
@@ -27,7 +28,8 @@ exports.handler = (event, context, callback) => {
 
 function process(s3Object, callback) {
     const configPath = path.resolve(__dirname, "config.json");
-    const processor  = new ImageProcessor(s3Object);
+    const fileSystem = new S3FileSystem();
+    const processor  = new ImageProcessor(fileSystem, s3Object);
```

Comment:
Why don't you `S3FileSystem` manage `s3Object` in its insance and get some data through it?